src/core/rowindex_array.cc

```diff
@@ -314,40 +313,47 @@ void ArrayRowIndexImpl::compactify()
   _resize_data();
 }
 
+template<typename TI>
+void get_next_index_to_skip(const TI* inputs, size_t& j)
+{
+  while (inputs[j] == inputs[j+1]) {
```

Comment:
This doesn't check that `inputs[j+1]` is within the array, so it may segfault for certain inputs.

I'd suggest to add a test where the user deletes a list of all identical indices (e.g. `[3]*1000`). This could be parametrized for lists of various lengths.

Reply:
good point, I agree.